backend/src/shortest-path/shortest-path.controller.ts

```diff
@@ -36,11 +36,11 @@ export class ShortestPathController {
       throw new BadRequestException();
     }
 
-    if (!query.startNode) {
+    if (query.startNode == null) {
       throw new BadRequestException();
     }
 
-    if (!query.endNode) {
+    if (query.endNode == null) {
```

Comment:
Always use strict equality comparison. This will fail in the CI tests.
Better use either
`!query.endNode && query.endNode !== 0)`
or
`query.endNode === null || query.endNode === undefined`